### PR TITLE
roe-1548 fix to date errors

### PR DESCRIPTION
**Pull request by okoye-companieshouse**
### JIRA link

https://companieshouse.atlassian.net/browse/ROE-1548

### Change description

Bug Fix to date field

### CODE REVIEW

File: src/validation/custom.validation.ts [hunk 1]
```diff
@@ -250,19 +246,71 @@ export const checkDateOfBirth = (dayStr: string = "", monthStr: string = "", yea
...
+  return true;
+};
+
 export const checkDateOfBirthFieldsArePresent = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
-  if (dayStr === "" && monthStr !== "" && yearStr !== "") {
-    return false;
-  } else if (monthStr === "" && dayStr !== "" && yearStr !== "") {
-    return false;
-  } else if (yearStr === "" && dayStr !== "" && monthStr !== "") {
+  if (dayStr === "" || monthStr === "" || yearStr === "") {
```
Comment (SteveHicksCH) [inline src/validation/custom.validation.ts:274]: reuse function checkAllDateFieldsArePresent

**Review by SteveHicksCH — COMMENTED** [review]
just two minor comments

File: src/validation/custom.validation.ts [hunk 2]
```diff
@@ -250,19 +246,71 @@ export const checkDateOfBirth = (dayStr: string = "", monthStr: string = "", yea
...
+  if (!checkAllDateFieldsArePresent(dayStr, monthStr, yearStr)) {
     return false;
   } else {
     checkMoreThanOneDateOfBirthFieldIsNotMissing(dayStr, monthStr, yearStr);
   }
   return true;
 };
 
+export const checkAllDateFieldsPresent = (dayStr: string = "", monthStr: string = "", yearStr: string = "") => {
+  if (dayStr === "" && monthStr === "" && yearStr === "") {
```
[the remark this answers is not in the document]
Comment (okoye-companieshouse) [reply src/validation/custom.validation.ts:283]: I have renamed the methods to an appropriate name now "checkDateFieldsForErrors"

File: src/validation/custom.validation.ts (hunk 1 above)
Comment (okoye-companieshouse) [reply src/validation/custom.validation.ts]: Amended

**Review by SteveHicksCH — APPROVED** [review]

**Pull request merged**